CMakeLists.txt

```diff
@@ -33,6 +33,31 @@ TRIBITS_PACKAGE_DECL(Kokkos) # ENABLE_SHADOWING_WARNINGS)
 # HAVE_KOKKOS_DEBUG.  We define KOKKOS_HAVE_DEBUG here instead,
 # for compatibility with Kokkos' Makefile build system.
 
+if (TPL_ENABLE_CUDA)
+  if (DEFINED CUDA_VERSION)
+    if (CUDA_VERSION VERSION_GREATER_EQUAL "7.5")
+      set(KOKKOS_HAVE_CUDA_GEQ_75 TRUE)
+    endif()
+    if (CUDA_VERSION VERSION_GREATER_EQUAL "8.0")
+      set(KOKKOS_HAVE_CUDA_GEQ_80 TRUE)
+    endif()
+  endif()
+endif()
+
+set(Kokkos_ENABLE_Cuda_Lambda_DEFAULT OFF)
+if (KOKKOS_HAVE_CUDA_GEQ_75)
+  if (CMAKE_CXX_FLAGS MATCHES "-expt-extended-lambda")
```

Comment:
Matches or contains?

Reply:
CMake doesn't have `CONTAINS`, `MATCHES` with a string as a regular expression essentially implements "contains". Also this has the nice property that `--expt-extended-lambda` should also be detected (two leading dashes).

Reply:
Oops, this option is likely version specific.  For example, CUDA 9.5 may not need the option.
Option detection should be tied to version or set/range of versions.

Reply:
well, right now its inside a check for version >= 7.5.
NVIDIA have not release a version yet that doesn't need this flag, so it shouldn't be too bad to wait until they do to add then "end" half of the range.

Reply:
Flag is there for 7.5, 8.0, and 9.0.